Update Analysis Service and API Management Outputs

senavar opened the pull request:
# Change

***Feel free to remove this sample text***
>Thank you for your contribution !

Update the output names for the Analysis Service and API Management modules 

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

eriqua commented on arm/Microsoft.ApiManagement/service/.bicep/nested_authorizationServers.bicep:
```diff
@@ -100,10 +100,10 @@ module authorizationServer '../authorizationServers/deploy.bicep' = {
 
 @description('The name of the API management service authorization server')
-output authorizationServerName string = authorizationServer.outputs.authorizationServerName
+output name string = authorizationServer.outputs.authorizationServerName
```
Since we also changed the correspondent output names in /authorizationServers/deploy.bicep it should probably be

```suggestion
output name string = authorizationServer.outputs.name
```

eriqua commented on arm/Microsoft.ApiManagement/service/.bicep/nested_authorizationServers.bicep:
```diff
@@ -100,10 +100,10 @@ module authorizationServer '../authorizationServers/deploy.bicep' = {
 
 @description('The resource ID of the API management service authorization server')
-output authorizationServerResourceId string = authorizationServer.outputs.authorizationServerResourceId
+output resourceId string = authorizationServer.outputs.authorizationServerResourceId
```
```suggestion
output resourceId string = authorizationServer.outputs.resourceId
```

eriqua commented on arm/Microsoft.ApiManagement/service/.bicep/nested_authorizationServers.bicep:
```diff
@@ -100,10 +100,10 @@ module authorizationServer '../authorizationServers/deploy.bicep' = {
 
 @description('The resource group the API management service authorization server was deployed into')
-output authorizationServerResourceGroup string = authorizationServer.outputs.authorizationServerResourceGroup
+output resourceGroupName string = authorizationServer.outputs.authorizationServerResourceGroup
```
```suggestion
output resourceGroupName string = authorizationServer.outputs.resourceGroupName
```

MrMCake commented on arm/Microsoft.ApiManagement/service/products/deploy.bicep:
```diff
@@ -73,16 +73,16 @@ module group 'groups/deploy.bicep' = [for (group, index) in groups: {
 
 @description('The Resources IDs of the API management service product APIs')
-output productApisResourceIds array = [for productApi in apis: resourceId('Microsoft.ApiManagement/service/products/apis', apiManagementServiceName, name, productApi.name)]
+output apisResourceIds array = [for productApi in apis: resourceId('Microsoft.ApiManagement/service/products/apis', apiManagementServiceName, name, productApi.name)]
```
```suggestion
output apiResourceIds array = [for productApi in apis: resourceId('Microsoft.ApiManagement/service/products/apis', apiManagementServiceName, name, productApi.name)]
```

MrMCake commented on arm/Microsoft.ApiManagement/service/products/deploy.bicep:
```diff
@@ -73,16 +73,16 @@ module group 'groups/deploy.bicep' = [for (group, index) in groups: {
 
 @description('The Resources IDs of the API management service product groups')
-output productGroupsResourceIds array = [for productGroup in groups: resourceId('Microsoft.ApiManagement/service/products/groups', apiManagementServiceName, name, productGroup.name)]
+output groupsResourceIds array = [for productGroup in groups: resourceId('Microsoft.ApiManagement/service/products/groups', apiManagementServiceName, name, productGroup.name)]
```
```suggestion
output groupResourceIds array = [for productGroup in groups: resourceId('Microsoft.ApiManagement/service/products/groups', apiManagementServiceName, name, productGroup.name)]
```

MrMCake commented on arm/Microsoft.ApiManagement/service/products/readme.md:
```diff
@@ -36,11 +36,11 @@ The following resources are required to be able to deploy this resource.
-| `productName` | string | The name of the API management service product |
-| `productResourceGroup` | string | The resource group the API management service product was deployed into |
-| `productResourceId` | string | The resource ID of the API management service product |
+| `apisResourceIds` | array | The Resources IDs of the API management service product APIs |
```
```suggestion
| `apiResourceIds` | array | The Resources IDs of the API management service product APIs |
```

MrMCake commented on arm/Microsoft.ApiManagement/service/products/readme.md:
```diff
@@ -36,11 +36,11 @@ The following resources are required to be able to deploy this resource.
-| `productResourceGroup` | string | The resource group the API management service product was deployed into |
-| `productResourceId` | string | The resource ID of the API management service product |
+| `apisResourceIds` | array | The Resources IDs of the API management service product APIs |
+| `groupsResourceIds` | array | The Resources IDs of the API management service product groups |
```
```suggestion
| `groupResourceIds` | array | The Resources IDs of the API management service product groups |
```

eriqua reviewed (dismissed):
Hi @senavar last thing if you can. Could you please try running the API Management pipeline from your _users/senavar/outputRefactor_ branch and attach the pipeline status badge in the PR description? Thanks!

MrMCake commented:
> users/senavar/outputRefactor

Took the liberty: [![ApiManagement: Service](https://github.com/Azure/ResourceModules/actions/workflows/ms.apimanagement.service.yml/badge.svg?branch=users%2Fsenavar%2FoutputRefactor)](https://github.com/Azure/ResourceModules/actions/workflows/ms.apimanagement.service.yml)

senavar commented:
> > Hi @senavar last thing if you can. Could you please try running the API Management pipeline from your users/senavar/outputRefactor branch and attach the pipeline status badge in the PR description? Thanks!
> 
> Took the liberty: [![ApiManagement: Service](https://github.com/Azure/ResourceModules/actions/workflows/ms.apimanagement.service.yml/badge.svg?branch=users%2Fsenavar%2FoutputRefactor)](https://github.com/Azure/ResourceModules/actions/workflows/ms.apimanagement.service.yml)
> 
> A couple tests failed. And with @rahalan changes we also have a bunch of conflicts to resolve. Maybe @rahalan can help you out with that

So I looked into the errors and its rather odd. It is saying that the readme output does not have all the outputs in the template file. Which is unexpected, as the readme script was ran for all files using the For-Each loop. Also checked the culprit readme and template file side-by-side. I don't see anything wrong. Can one of you also do a sanity check and make sure I'm not going crazy? @MrMCake  @eriqua

MrMCake commented:
Hey @senavar, I fixed the issue. The problem was as follows:
By renaming the `resourceId` output in the `products` template to just `resourceId`, the also used function `resourceId()` could not be used anymore (i.e. bicep is unable to differentate the output name from the function (duh)). I updated it now to a different implementation that returns the ID from the module deployment outputs (api + groups) directly.

senavar commented:
> Hey @senavar, I fixed the issue. The problem was as follows: By renaming the `resourceId` output in the `products` template to just `resourceId`, the also used function `resourceId()` could not be used anymore (i.e. bicep is unable to differentate the output name from the function (duh)). I updated it now to a different implementation that returns the ID from the module deployment outputs (api + groups) directly.

Thanks! Will make note of this as I work through other modules and adjust as needed. Looks like the pipeline is now green and all good to go for merging :)
